src/OnChain.ts

```diff
@@ -171,6 +178,10 @@ export const OnChainMixin = (superclass) =>
 
           // normal onchain payment path
 
+          if (amount < yamlConfig.onchainDustAmount) {
```

Comment:
we don't want to put this in the lock. it's better to throw prior to the lock if we throw and acquiring the lock is not needed to know if this condition is true or not

Reply:
I'm doing it inside the lock because that's where we determine whether it's on onchain_on_us or a regular onchain payment. The intention being that I want to allow a dust amount if it's an `onchain_on_us` but no if it's `onchain`.

Reply:
yup you're right

Reply:
Can the `onchainDustAmount` be passed in as constructor config instead of read directly from `yaml`? That would make the code more modular and easier to test.